keras/callbacks.py

```diff
@@ -779,32 +785,46 @@ def is_indexed_slices(grad):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        if self.embeddings_freq:
+        if self.embeddings_freq and self.embeddings_data is not None:
             embeddings_layer_names = self.embeddings_layer_names
 
             if not embeddings_layer_names:
                 embeddings_layer_names = [layer.name for layer in self.model.layers
                                           if type(layer).__name__ == 'Embedding']
+            self.embeddings = []
```

Comment:
just a matter of style and not so important if things work, but lot's of variables from here on have names involving "embedding" - do we really need that many? And if so, are more expressive names conceivable?